From: "Y.C. Chen" <yc_chen@aspeedtech.com>
To: dri-devel@lists.freedesktop.org
Cc: yc_chen@aspeedtech.com, kuohsiang_chou@aspeedtech.com, eich@suse.com,
        airlied@redhat.com
Subject: [PATCH 1/2] drm/ast: open key before detect chips
Date: Wed, 10 Sep 2014 12:07:53 +0800

From: "Y.C. Chen" <yc_chen@aspeedtech.com>
References: bnc#895983
Patch-mainline: submitted for upstream
Signed-off-by: Egbert Eich <eich@suse.de>

Some config settings like 3rd TX chips will not get correctly
if the extended reg is protected

Signed-off-by: Y.C. Chen <yc_chen@aspeedtech.com>
Reviewed-by: Egbert Eich <eich@suse.de>
---
 drivers/gpu/drm/ast/ast_main.c |    2 ++
 1 file changed, 2 insertions(+)

--- linux-3.0-SLE11-SP3.orig/drivers/gpu/drm/ast/ast_main.c
+++ linux-3.0-SLE11-SP3/drivers/gpu/drm/ast/ast_main.c
@@ -68,6 +68,8 @@ static int ast_detect_chip(struct drm_de
 	struct ast_private *ast = dev->dev_private;
 	uint32_t data, jreg;
 
+	ast_open_key(ast);
+
 	if (dev->pdev->device == PCI_CHIP_AST1180) {
 		ast->chip = AST1100;
 		DRM_INFO("AST 1180 detected\n");
